### DISCUSSION TITLE
.net core support for Authentication

### DISCUSSION THREAD

**Discussion by renszzzzzzz** [microsoft/FluidFramework#11281] [Q&A] [answered]
Hello, I am trying to implement AzureFunctionTokenProvider() described here:  [Authentication and authorization in your app](https://docs.microsoft.com/en-us/azure/azure-fluid-relay/concepts/authentication-authorization) and [How to: Write a TokenProvider with an Azure Function](https://docs.microsoft.com/en-us/azure/azure-fluid-relay/how-tos/azure-function-token-provider). 

My current backend is Azure Functions using .NET (C#), however there is currently no fluid-framework support. Is it possible to provide documentation on how to support custom tokenProviders outside the InsecureTokenProvider() and AzureFunctionTokenProvider().

**Comment by skylerjokiel** [comment, marked as answer]
@renszzzzzzz , token generation is not restricted to javascript. The only requirement of the service is to generate a jwt token. You can use any backend language to do this. You can learn about the token structure here:
https://learn.microsoft.com/en-us/azure/azure-fluid-relay/concepts/authentication-authorization
